cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/feature/DefaultFeatureFlagProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2014-2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.internal.app.feature;
+
+import io.cdap.cdap.api.feature.FeatureFlagsProvider;
+import io.cdap.cdap.common.conf.Configuration;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DefaultFeatureFlagProvider implements FeatureFlagsProvider {
+
+  private final Configuration configuration;
+  public DefaultFeatureFlagProvider(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  /**
+   * Get {@link Map} of properties prefixed with FeatureFlagsProvider.FEATURE_FLAG_PREFIX and enabled values
+   * Feature flag prefix in the mapping are trimmed to remove the prefix
+   *
+   * @return mapping of feature flag properties from configuration with prefix stripped.
+   */
+  public Map<String, Boolean> getFeatureFlags() {
+    return configuration.getPropsWithPrefix(FeatureFlagsProvider.FEATURE_FLAG_PREFIX).entrySet()
+      .stream()
+      .collect(Collectors.toMap(Map.Entry::getKey, (entry) -> Boolean.getBoolean(entry.getValue())));
```

Comment:
The `Boolean.getBoolean` is reading from the system properties, are you sure that's what we wanted? I believe you should be using `Boolean.parseBoolean` instead?